### PR TITLE
PCSX2-Counters: Revert to older VRender/VBlank timings

### PR DESCRIPTION
**Pull request by lightningterror**
This fixes timing issues Jak II, Fatal Fury Battle Archives (1) and
Shadow Of Rome.

Follow up from #2275

I tested Shadow of Rome ntsc demo and it does fix the fmv in the intro.

Close: #360
Close: #1712
Close: #2135

### CODE REVIEW

**Comment by lightningterror** [comment]
I don't know if any other changes are required, such as the changes in m_resetVsyncTimers.
Anyone can finish up the pr if some extra changes are needed.

File: pcsx2/Counters.cpp [hunk 1]
```diff
@@ -202,6 +202,10 @@ static void vSyncInfoCalc(vSyncTimingInfo* info, Fixed100 framesPerSecond, u32 s
 
 	u64 Blank = HalfFrame / 2; // PAL VBlank Period is off for roughly 22 HSyncs
```
Comment (ramapcsx2) [inline pcsx2/Counters.cpp]: Would be nice if you edited this please :)

Comment out the whole line please, and replace the original comment with "broken code, used before 01.02.2019" (or similar).
Then just use the correct line below for u64 Blank = ..

**Comment by lightningterror** [comment]
Looks like a job for @MrCK1 and @RedPanda4552 to do some testing.

File: pcsx2/Counters.cpp [hunk 2]
```diff
@@ -200,7 +200,12 @@ static void vSyncInfoCalc(vSyncTimingInfo* info, Fixed100 framesPerSecond, u32 s
 	// One test we have shows that VBlank lasts for ~22 HBlanks, another we have show that is the time it's off.
 	// There exists a game (Legendz Gekitou! Saga Battle) Which runs REALLY slowly if VBlank is ~22 HBlanks, so the other test wins.
 
-	u64 Blank = HalfFrame / 2; // PAL VBlank Period is off for roughly 22 HSyncs
+	// Broken code, see https://github.com/PCSX2/pcsx2/pull/2835
+	// u64 Blank = HalfFrame / 2; // PAL VBlank Period is off for roughly 22 HSyncs
+
+	// Alternative vsync timing.
+	// This fixes timing issues Jak II, Fatal Fury Battle Archives (1) and Shadow Of Rome.
+	u64 Blank = (Frame / scansPerFrame) * (gsVideoMode == GS_VideoMode::NTSC ? 26 : 22);
```
Comment (ramapcsx2) [inline pcsx2/Counters.cpp]: Perfect, thanks :)

**Comment by RedPanda4552** [comment]
No major differences in the batch I picked. I think the slight desync I was noticing (running two PCSX2's at once) was due to the usual controller latency (one instance gets the input, then the other a moment after).
The only thing of actual significance I noticed was that Final Fantasy 8 (PS1), seemed to have a slight FMV speedup, but this isn't saying much considering FMVs are utterly broken in there.

**Comment by ramapcsx2** [comment]
The change probably has more effect on FMV than actual gameplay (noticed this in testing).
It's likely that FMV sync looks more at VBlank timing than other code.
That said, it does look good :) Thanks for testing!

**Comment by atomic83GitHub** [comment]
This PR still need some testing? I can try it on various games.

**Comment by lightningterror** [comment]
Yes more tests are welcome.

**Comment by lightningterror** [comment]
If we can't do enough testing for 1.6 then we can leave it for 1.8 release.

**Comment by MrCK1** [comment]
I can do more to ight, it will be fine IMO

**Comment by ramapcsx2** [comment]
I recommend not putting this into 1.6. 
While I'm pretty sure that it's correct, there could be issues that appear only with this corrected timing.

**Comment by atomic83GitHub** [comment]
I tested many games with this PR and not a single regression to note, there is even an improvement with a .elf named "PicoDrive" it is much faster with the new Vsync (especially seeable with VirtuaRacing).

**Comment by ramapcsx2** [comment]
Yep, I believe that most classic collections (that use an emulator for old systems) will work better, see Phantasy Star 4 (also fixed with this change) :)

**Comment by ssakash** [comment]
I suggest doing some benchmarks comparing the PS2 timing behavior before merging this, this is a really crucial subroutine where minor discrepancies could cause games to go havoc.

**Comment by ssakash** [comment]
```
1/ Persona - 3 (NTSC)
New Code:
FrameRate : 2997  , Render: 2460061 , Blank: 2460060
HRender : 9354   , Hblank : 9353 , HScanlinsperFrame : 526
hSyncError: 180
---------------------------------------------------------------
2/ ICO - 60HZ video mode (progressive)
New Code:
FrameRate : 2997  , Render: 2460061 , Blank: 2460060 (Render + Blank (AKA Frame) = 4920121)
HRender : 9354   , Hblank : 9353 , HScanlinsperFrame : 526
hSyncError: 180

Old Code:
FrameRate : 2997  , Render: 4433721 , Blank: 486399 (Render + Blank (AKA Frame) = 4920120)
HRender : 9354   , Hblank : 9353 , HScanlinsperFrame : 526
hSyncError: 179
----------------------------------------------------------------
3/ ICO - 50HZ video mode (progressive)

New Code:
FrameRate : 2500  , Render: 2949121 , Blank: 2949120
HRender : 9393   , Hblank : 9392 , HScanlinsperFrame : 628
hSyncError: -249

Old Code:
FrameRate : 2500  , Render: 5484988 , Blank: 413252
HRender : 9393   , Hblank : 9392 , HScanlinsperFrame : 628
hSyncError: -250
-----------------------------------------------------------------
4/ Shadow of Rome (NTSC)

New Code:
FrameRate : 2997  , Render: 2460061 , Blank: 2460060
HRender : 9354   , Hblank : 9353 , HScanlinsperFrame : 526
hSyncError: 180

Old Code:
FrameRate : 2997  , Render: 4433721 , Blank: 486399
HRender : 9354   , Hblank : 9353 , HScanlinsperFrame : 526
hSyncError: 179
```

**Comment by ssakash** [comment]
Also wouldn't this neglect the 50/50 proportion of the render and blank which was supposed to be there according to Jake? so I'm not too sure if this code is accurate or not.

**Comment by ssakash** [comment]
Really sorry for jumping out the blue, just that my free time is ultra limited these days and I try to help whenever I can. xD

**Comment by ramapcsx2** [comment]
We don't have enough details on that 50/50 assumption. It could be wrong, or only partially correct, depending on what it is applied to.
With this much uncertainty (it's just a comment, no tests or references), I'd rather ignore it and implement video timings according to basic standards.
(IE: I'd prefer short blanking + long render instead of 50 / 50. The one good test we have agrees.)

Could you check your numbers?
It appears as if the old code / new code results are swapped.
The new code should produce a VBlank that's significantly shorter than VRender.

**Comment by ssakash** [comment]
@ramapcsx2 

That testcase was taken like a year ago and unless I'm wrong, this new code in the PR is actually the older version of render/blank calculation.

**Comment by ramapcsx2** [comment]
Okay, all is well then :p

**Comment by atomic83GitHub** [comment]
For the upcoming 1.6 release, why not add a new "Hack" like this to avoid regression, it makes Jak 2 playable so I think it's a good idea not to wait ver 1.8:

- Old Vsync timing: switch back to the old Vsync solution. Use it when your game is running very slow or too fast.

**Comment by lightningterror** [comment]
I'd prefer if we don't add more gui options since we have plenty already.

**Comment by atomic83GitHub** [comment]
Hidden in the ini maybe?

**Comment by ramapcsx2** [comment]
It isn't necessary to toggle this code as it is correct and the old is wrong.
The only reason it shouldn't be merged yet is that it potentially affects many games in unknown ways, could trigger other issues and so on.
It's just hard to do a full coverage test of this, so it's best left for a later release.

**Comment by atomic83GitHub** [comment]
This commit broke "Legendz Gekitou! Saga Battle" The changes you reverted fixes that game from runnig at a very poor framerate.

**Comment by ramapcsx2** [comment]
Ok, pretty sure this is another fault in Legendz Gekitou that gets triggered by the corrected timings.
Good to know there is a game to compare though.

**Comment by lightningterror** [comment]
Good thing we didn't merge it yet then.

**Comment by ZwipZwapZapony** [comment]
(Possibly noteworthy note: This pull request essentially undoes commit c8a7a70.)

Merging this pull request is not perfect, as it negatively impacts at least one game (Legendz Gekitou! Saga Battle).
Not merging this pull request is also not perfect, as it fixes something with at least 3 games (Fatal Fury Battle Archives, Jak II, Shadow of Rome).
If we can't have it perfect either way, I'd prefer to have a "per-game fix" that handles this for games that benefit from it, alike (the closed) pull request #2275, so that we can still have the best of both worlds until a "perfect" solution is envisioned.

As it stands at the moment, Jak II is _playable_, as it can be _played_... but it's very hard (for me) to _enjoy_ it with its "random" internal game speed and framerate patterns.
(From my own limited experience, increasing the EE cyclerate in PCSX2 can lessen/prevent this problem... but then PCSX2 itself takes more processing power, causing PCSX2 as a whole to slow down for me. Decreasing the EE cyclerate seems to lessen/prevent the game speed problem too, but the internal game framerate gets lowered, which is not desireable for me.)
A "per-game fix" for this pull request would make Jak II enjoyable for me and likely many others, from what I can tell. So... why not, exactly?

~

TL;DR: This commit breaks a game, but fixes others. Why not have it as a "per-game fix", to have the best of both worlds depending on the game?

**Comment by lightningterror** [comment]
@ZwipZwapZapony it was discussed on discord that the issue popping up with Legendz Gekitou! Saga Battle might be another old timing issue that just started to show up because of this change. Rama said that this change is more accurate and the best solution would be to investigate what the issue with Saga Battle is.

**Comment by CallistoNTG** [comment]
This PR appears to alleviate or even potentially fix https://github.com/PCSX2/pcsx2/issues/1325 The game still has streaming problems with the hardware renderers, but I booted the game running in SW mode, and the opening cutscene where Harry looks for the flashlight in the crashed car works correctly, or close to correctly.

**Comment by atomic83GitHub** [comment]
This PR makes Popcap hits games tiny better. They have less stuttering when doing combos.

**Comment by lightningterror** [comment]
@ramapcsx2 there was talk to actually merge this for 1.6 release instead of 1.8 on discord, what are your thoughts?

**Comment by ramapcsx2** [comment]
It's risky.
Make sure to give it lots of testing first..

**Comment by tadanokojin** [comment]
LT can you rebase this to master so we can get this to our testers.

**Comment by turtleli** [comment]
You can just press the Re-build PR button on Appveyor to generate a build of the PR merged with master (as long as there are no merge conflicts).

**Comment by tadanokojin** [comment]
Ah, I didn't know it was against master. Thanks Li.

**Comment by ramapcsx2** [comment]
Well, maybe we can do a vote :p 

Do we want to test and include this PR in 1.6, or wait until after the release?

Please leave a thumbs up for including, and a thumbs down for waiting.
If you can't decide, leave some Eyes ;p

**Comment by refractionpcsx2** [comment]
This needs through testing before it's merged since vsync timing is so sensitive.  I wouldn't feel comfortable in merging this for 1.6 without having a good test on lots of games.

**Comment by ramapcsx2** [comment]
Okay, leave a dislike then please :p 
I voted for waiting as well, same reason as you.

**Comment by Dotty** [comment]
Well, in that case it should have been voted before merging beforehand. This would just be reverting a previous change. Which caused regressions in several games. Just because it hackishly fixed one. It's kind of silly to vote only to undo something like this when it wasn't voted beforehand.
That this is the one change left for making Jak 2 playable, as of the recent fixes to eyes and textures, is a pretty notable factor too.
Just after 1.6 would be neat too though, absolutely. Appveyor gives builds for anyone who wants them for now.

**Comment by PSI-Rockin** [comment]
Yes, the previous change should have been voted on before merging. This, too, should also be voted on. 1.5 is currently in a feature freeze, and messing with timings is a guaranteed way of introducing new regressions, which is not what should happen during a freeze. It doesn't matter if the new timings are more accurate: unless you have a 100% cycle-accurate emulator (we don't), fixing timings for one game breaks timings for another game. If you absolutely must have a fix for Jak now, then use Appveyor like you suggested it, or compile it yourself.

**Comment by willkuer** [comment]
Honestly how can you ensure in a more efficient way that stuff is tested than by just merging?

See recent pthread changes. They were merged, regression spottet, reverted.

If you would‘ve merged this PR a few month ago you would already know whether there are regressions or not. The emitted pre-releases are not guaranteed to be bug free.

As long as you have no QA engineers and fully automated testsuites I don‘t see how you can guarantee a higher quality by leaving this branch unmerged with the comment „I feel this needs more testing“.

**Comment by lightningterror** [comment]
Voting to leave it for 1.8 release. Seems the safest, especially now this close to 1.6 release.

**Comment by arcum42** [comment]
Yeah, my thought is that we release 1.6, and then merge it in afterwards, so that it gets the testing it needs, but in the development branch, rather than having to do 1.6.1 to revert it or something if things don't go well.

**Comment by lightningterror** [comment]
Merging it mid 2018 would've been enough time I guess to gather sufficient data for any potential issues.

**Comment by tadanokojin** [comment]
Please at least allow testers to at least look at it and give us an idea of the state before we made a decision.

**Comment by ssakash** [comment]
@ramapcsx2  IIIRC you had a ELF testcase for monitoring the render/blanks timing in comparison to the PS2 behavior, did you give this commit a shot with that?

**Comment by atomic83GitHub** [comment]
Honesty, the first idea was much better, adding a parametre to switch from the ld to the new vsync for the games that need it, no regressions and fixes the games that need the new one.

**Comment by CallistoNTG** [comment]
>Honesty, the first idea was much better, adding a parametre to switch from the old to the new vsync for the games that need it in the GameDB, no regressions and fixes the games that need the new one.

I disagree. You don't make the inaccurate behavior the default and then enable the accurate behavior selectively. You make the accurate behavior the default, and selectively disable it for known regressions, if they exist, until their cause can be determined/fixed. The thing about accuracy improvement is that most people have no clue how games are supposed to behave, and they will never bother turning on the setting or documenting that it fixes some subtle pacing issue 10 hours into a 50 hours long game. You don't know which games benefit from this accuracy improvement. Some people seem fixated on Jax II, but I think that you'll likely find it improves streaming and timing accuracy across the board in subtle and not so subtle ways.

You end up with situations like Project 64, where they enabled a very bad 32 bit hack for a stupidly marginal performance gain by default. And then disabled it for games which were visibly broken. And guess what? People just assume the broken behavior is intended because they haven't seen the real thing in a decade or more, if ever. So games stay broken, and nobody knows. That doesn't lead an emulator towards being more accurate and compatible. It causes stagnation.

**Comment by tadanokojin** [comment]
> adding a parametre to switch from the old to the new vsync for the games that need it in the GameDB

I'm against it. I looked at the data Rama has provided so far and it seems convincing that this is the more accurate solution. For reference you can check #2275.

Broken games with fixed vsync timings means that there are issues elsewhere.
No sense hiding that fact behind a flag in gamedb. Otherwise we rely on broken behavior 99% and accurate flag for 1%.

Whether we should do it for 1.6.0 or 1.8.0 is another question. I'd personally love it for 1.6.0 if circumstances allow but I'd like to get some more information about what sort of effect this has on compatibility whether we do or don't.

I just hate to see a good PR like this sit.

**Comment by Dotty** [comment]
Speaking of fixation on Jak II, I've put 10 hours into that on this pull so far, reaching "73%" (90% of the story done in fair detail). Not a single random speed burst. So it certainly fixes that.
(Though there are still other problems. Like frequent crashing with mipmaps on or infrequent crashing otherwise. And a few scenes with messed up eyes still (Torn near the begining with black empty eyes, and Samos in Kiera's garage with static eyes). Those are out of the scope of this pull, of course.)
Still definitely playable. Seemingly from start to finish. Which is amazing for such a troublesome game. So thanks.

**Comment by atomic83GitHub** [comment]
I am in favor of 1.6, Jak games receive multipnes fixes and this one is crucial to make Jak 2 playable, Legendz can be fixed later actually I think.

**Comment by ramapcsx2** [comment]
@Dotty 
A previous merge that would be undone?
I seem to have missed that. As far as I know, the regular "VBlank = half a frame" timing is still in?

@CallistoNTG 
I fully agree.
This PR should never be an option either. It's the correct behaviour for this part of the timing calculations.
But once this is merged, we need to find out what actually breaks the few known (so far) titles.

So yeah, lots of votes! :)
I think we need to clear up what's going on with those Jak changes first. 
How do they relate to this PR?

**Comment by refractionpcsx2** [comment]
Something that was pointed out to me, this is just reverting the code to what I had before it was reverted to fix 2 games of unknown source as you can see in the original revert
https://github.com/PCSX2/pcsx2/commit/c8a7a709dcc33172bc252fb6ac7f2fb75da65605#diff-57990eb80cdbbaa83770efa035b5fbe2

**Comment by ramapcsx2** [comment]
Oh, that's really far back then.
Is this the commit @Dotty spoke of?

In any case, today the situation has changed a bit.
We have a test that says the VBlank = 20 something lines, and we have a bunch of games that work better with it as well.
Looking at the test visuals, they look just right. Compared to the way too long VBlank time with current code, the PR code is almost a perfect match to real hardware.

So that makes me pretty certain that the change is correct :p

**Comment by KrossX** [comment]
Merge it already. It's about time, according to the file change history. =P

**Comment by Dotty** [comment]
Yes it is. Apologies for my confusion, seeing that that was just a revert as well.
As for Jak fixes (https://github.com/PCSX2/pcsx2/pull/3001 https://github.com/PCSX2/pcsx2/pull/3018), they don't really directly relate to this pull in any other way than fixing the game/s. But as of all 3, they are considerably playable.

**Comment by refractionpcsx2** [comment]
The question I have is, what we're those two games and have they been tested? Do you remember rama?

**Comment by ramapcsx2** [comment]
No, I don't remember what they were. It's likely that one was the Sega Ages Phantasy Star Collection, so basically Mega Drive emulation.

All those titles (that use emulation) are a good start to test timing changes.
And yeah, it looks like this PR is good for them, fixing PCSX2 slowdowns (that don't happen on real hardware).

I think it may also be necessary to test this PR with different video modes (progressive / interlaced), when the games allow those options natively.
Sega Ages titles do! :)

**Comment by ramapcsx2** [comment]
So.. What do we do now?
At this time there are 10 votes for including it in 1.6 vs 8 votes to wait.

I've read everyone's opinions and it seems that all except one favour waiting, with the reasoning that introducing a timing change shortly before release is a bad idea.

The comment that favours including it reasons that it's a good change and it shouldn't stay unmerged forever.
I agree with that, but I'd still like to do it right after 1.6 is out.

As for all the other votes, it would be nice if you gave a comment on your choice.

**Comment by RedPanda4552** [comment]
My vote was wait until after 1.6. Reason, we are supposed to be in feature freeze to prevent introduction of bugs. Unless someone is ready to test half the PS2 library on this, we need time. We don't know what this does for playability. We need people to do things like run through Ratchet and Clank, Xenosaga, SotC, etc. We need full playthroughs. We need a mountain of test cases saying "yes, this doesn't affect playable statuses". Unless we have that, this needs to be on the 1.7 development series where no one will be upset if we break something for a while.

Though I agree with Rama. This PR is so stale that if we wait until after 1.6 release, this needs to be one of the first PRs to merge.

**Comment by tadanokojin** [comment]
I've heard some mumblings from CK and other testers that they haven't found issues with it yet in their games. Still hasn't been tested very thoroughly, though.

If we wait, it will be another stable release before Jak 2 is really playable. If we merge now then we have the whole jak trilogy (including eyes and shadow fixes) with the only outstanding issues being mipmapping related (unless there's another I'm not thinking of).

Obviously the risk there is potentially we merge something that might not be ready. So I guess it depends on your poison.

How many games does this have to work with before we consider it "ready"?

**Comment by prafullpcsx2** [comment]
I have tried many games with this. Can't find anything broken. I think it should be merged immediately. This has already been pending for 7 months. If we can't test enough games to merge it now we won't be able to test enough games even for 1.8 release and this will stay pending forever.

**Comment by atomic83GitHub** [comment]
I tested that on several homebrew softwares as well and no sign of broken stuff

**Comment by lightningterror** [comment]
It has my approval for merging for 1.6 based on recent discussion with rama and turtleli, plus seems that it's been tested properly on many games already.

**Comment by turtleli** [comment]
> based on recent discussion with rama and turtleli

Maybe just me, but I can't find anything on Discord that I've said about this that has much significance (I've mentioned its been reverted back and forth more than once, but that's about it).

Regardless of whether this is merged for or after the 1.6 release, I think the commit message and code comments need to be changed. The comments and code actually seem mismatched, more so if you look at the history of the code.

**Comment by lightningterror** [comment]
Was referring to what you said about this code making more sense than what's currently on master.

Anyway anyone feel free to suggest or change the commit description and code comments to something more fitting.

**Comment by ssakash** [comment]
Anyone ran tests on games with 480p videomode?

**Comment by ramapcsx2** [comment]
Yep, but just one game. There were no differences at all between 240p (or 480i) and 480p.
If we're lucky this means that the output resolution doesn't affect the EE side video mode.

**Comment by mirh** [comment]
I don't have any special opinion either way, but considering what [big change](3090) is supposed to land for 1.6, we could as well throw in as many need-to-test features as possible at this point :smile:

**Comment by ramapcsx2** [comment]
Well, I couldn't find any more issues with this. The votes mostly favour merging for 1.6.

Fine by me then :)

**Comment by lightningterror** [comment]
I won't be the one merging this, I'll pass on that :p

**Comment by turtleli** [comment]
I'll look into updating the comments.

**Comment by turtleli** [comment]
> The question I have is, what were those two games and have they been tested?

Smackdown vs Raw 2006 and Portal Runner according to https://forums.pcsx2.net/Thread-vSyncInfoCalc-timing-change.

**Comment by refractionpcsx2** [comment]
Good find, I seemed to mention Dynasty Warriors too, can't remember which or what problem it was lol

**Comment by Evil-Moo** [comment]
The only Dynasty Warriors thing related to this issue I'm aware of is this: https://github.com/PCSX2/pcsx2/issues/2135

This updated timing fixes a save corruption issue (apparently some kind of anti-cheat mechanism being tripped without the fix).

**Comment by refractionpcsx2** [comment]
Ah that could be it! Thanks @Evil-Moo

**Comment by turtleli** [comment]
I've updated the comments to reflect the code and commit history better.

I also changed
`u64 Blank = (Frame / scansPerFrame) * (gsVideoMode == GS_VideoMode::NTSC ? 26 : 22);`
to
`u64 Blank = Scanline * (gsVideoMode == GS_VideoMode::NTSC ? 26 : 22);`
which is the same thing.

**Comment by tadanokojin** [comment]
Thanks Li,

Am I understanding right that Smackdown vs Raw 2006 and Portal Runner still need testing?

**Comment by turtleli** [comment]
I don't see anyone stating that they've tested those games, so it may or may not have been tested.

It would probably help if people did mention what games they tested and how much they tested it.

**Comment by prafullpcsx2** [comment]
Portal runner works but SMVR 2006 titantron is missing with this.

**Comment by refractionpcsx2** [comment]
@prafullpcsx2 how does SMVR 2006 run on master?

**Comment by prafullpcsx2** [comment]
On master, titantron works irrespective of ee cycle skipping.

**Comment by ramapcsx2** [comment]
Oh, that's interesting.   
The code decides that there isn't enough time to render everything and so it skips it?

**Comment by tadanokojin** [comment]
Here's some additional context from discord that might be insightful.

Prafull Today at 4:22 PM
Undertaker titantron is black on counters revert, white on master but doesn't work on either of them. It works on both with ee skipping 1 or higher

Prafull Today at 4:25 PM
HHH titantron is black on counters revert, works on master. Works on revert with ee skipping 2 or higher

**Comment by Dotty** [comment]
I tested a few games too. And ditto, nothing but Jak 2 stood out.
- Jak 1 - 2 hours
- Jak 2 - 16 hours (Almost done with the story)
- Jak 3 - 4 hours
- Jak X - 5 hours (On the third quarter now)
- Kuon - 1 hour (Sugoroku minigame still broken, no change; main game still works)
- SMT Nocturne - 1 hour
- SMT Digital Devil Saga 1 - 1 hour
- SMT Digital Devil Saga 2 - 20 minutes
- Persona 3 FES - 10 minutes in story/town, 20 minutes in dungeon
- Persona 4 - 20 minutes
- YuGiOh Duelists of the Roses - a few duels in 20 minutes

With Portal Runner and SvR2006, what was mentioned as broken there years ago? The thread seems to be hidden or removed to the public.

**Comment by MrCK1** [comment]
I tested the following games without issues, nothing stood out as other comments mentioned

- Mission Impossible
- Wipeout Fusion
- Hitman Contracts
- Guitar Hero 3
- Shadow of Zorro
- Syphon Filter: Dark Mirror/Logan's Shadow
- CoD: Finest Hour
- Shox
- The Suffering/Ties That Bind
- Test Drive
- Knights of the Temple I / II
- Minority Report
-  GTA VC/SA
- Eragon
- Max Payne

**Comment by lightningterror** [comment]
If we have sufficient testing data then we can go ahead with a merge I guess, Rama the honor is yours :)

**Comment by turtleli** [comment]
...I'll handle the merge. I redid the comments so it's probably better that they're somewhat traceable to me.

First, are we fine with merging? If I don't see any major objections then I'll merge it within 2-3 days. So far:
 * 1 game (WWE Smackdown VS Raw 2006) is worse off, but that game has issues on master as well. It can be worked around with EE skipping though.
 * Portal Runner works. This was apparently broken with the same timings but now works.
 * Legends Gekitou works fine from what I've read on Discord. I don't know its status on master.
 * A few games are improved - Dynasty Warriors 3, Jak 2, Shadow of Rome, Popcap hits, Phantasy Star 4, Silent Hill: Shattered Memories.
 * Most tested games are unaffected.

In short, PR seems to improve things.

**Comment by ramapcsx2** [comment]
Yep, I'm good with the amount of testing done.

**Comment by Krusher97** [comment]
This PR also fixed the vertical screen tearing in the FMVs of Darkwatch
